api/src/main/proto/stellarstation/api/v1/stellarstation.proto

```diff
@@ -0,0 +1,147 @@
+  // and returned. When the client is done with the stream, it should close it successfully. If the
+  // stream is cancelled with an unexpected error at any point, telemetry received before the next
+  // call to `OpenSatelliteStream` to resume the stream will be buffered up to 1 minute, and if no
+  // call to `OpenSatelliteStream` is made with `stream_id` specified to resume the stream, it will
+  // expire in 10 minutes and a call to `OpenSatelliteStream` with that `stream_id` will be closed
+  // with a `ABORTED` error.
+  //
+  // If the satellite is not found or the client is not authorized for it, the stream will be closed
+  // with a `NOT_FOUND` error.
+  rpc OpenSatelliteStream (stream SatelliteStreamRequest) returns (stream SatelliteStreamResponse);
+}
+
+// Request for the `OpenSatelliteStream` method.
+message SatelliteStreamRequest {
+  // The ID of the satellite to open a stream with. The ID of a satellite can be found on the
+  // StellarStation Console page for the satellite.
+  string satellite_id = 1;
+
+  // The `SatelliteStreamResponse.stream_id` from a previously opened stream to resume. If the
+  // specified stream has already expired or been closed, the stream is closed with a `ABORTED`
+  // error.
+  string stream_id = 2;
+
+  // A request to send on the satellite stream. Currently only command sending to the satellite is
+  // supported. The request does not need to be filled on the first request to
+  // `OpenSatelliteStream`.
+  oneof Request {
+    // A request to send commands to the satellite.
+    SendSatelliteCommandsRequest send_satellite_commands_request = 3;
+  }
+}
+
+// A request to send commands to a satellite.
+message SendSatelliteCommandsRequest {
+  // The framing to be applied to `command` before transmission. If `command` has already been
+  // framed, this should be set to RAW and no additional processing will take place before
+  // modulation. When applying framing, each `command` will be treated as a single frame. Framing
+  // parameters used will be the ones registered for the satellite in the StellarStation Console.
+  Framing output_framing = 1;
+
+  // The command frames to send to the satellite. All commands will be transmitted in sequence
+  // immediately, during which time telemetry will not be received. After all commands have been
+  // transmitted, telemetry receive will be immediately re-enabled. Them aximum size of each command
+  // is 2MB. If a command larger than 2MB is received, the stream will be closed with a
+  // `RESOURCE_EXHAUSTED` error.
+  repeated bytes command = 2;
+}
+
+// A type of framing of a binary payload used in satellite communicaation.
+enum Framing {
+  // No framing. For sent payloads, this is usually pre-framed data while for received ones, it is
+  // raw IQ data.
+  RAW = 0;
+
+  // AX.25 (Amateur X.25) framing - https://en.wikipedia.org/wiki/AX.25
+  AX25 = 1;
```

Comment:
If we specify AX25 during transmission, don't we need some mechanism to allow the user to set the desired AX25 address field, control field, and PID field? We could have reasonable defaults if they don't care, but really, the only fields that aren't user-specific here are the start/end flags and checksum fields.

Reply:
Yeah the parameters are registered into the DB, rather than being in the API, documented here

https://github.com/infostellarinc/stellarstation-api/blob/master/api/src/main/proto/stellarstation/api/v1/stellarstation.proto#L78

Ricardo thinks this should be fine for conceivable use cases, if something comes up to require it in the API we can add it later. Does that make sense?

Reply:
Oh, there it is. :) Makes sense.